src/ciflow-bot.ts

```diff
@@ -225,11 +231,15 @@ export class CIFlowBot {
     ).upsertRootComment();
   }
 
-  async setLabels(): Promise<void> {
-    if (
+  hasLabelBeforePR(): boolean {
```

Comment:
Nit (plural labels vs singular label)
```suggestion
  hasLabelsBeforePR(): boolean {
```